# Case Study on Testing of Hypothesis - #04

A company started to invest in digital marketing as a new way of their product promotions. For that they collected data and decided to carry out a study on it.

● The company wishes to clarify whether there is any increase in sales after
stepping into digital marketing.

● The company needs to check whether there is any dependency between the
features “Region” and “Manager”.

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats
from scipy.stats import chi2_contingency

In [3]:
df=pd.read_csv('Sales_add.csv')

In [4]:
df.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Month                           22 non-null     object
 1   Region                          22 non-null     object
 2   Manager                         22 non-null     object
 3   Sales_before_digital_add(in $)  22 non-null     int64 
 4   Sales_After_digital_add(in $)   22 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 1008.0+ bytes


In [6]:
df.describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

# Insights

We have the data of 5 month sales details before and after digital marketing

From the above we can analyse before digital marketing there is only the average of 149239.95 sales and it is increased to 231121.7 after the digital marketing


# 1. The company wishes to clarify whether there is any increase in sales after stepping into digital marketing

# One tailed Paired Samples T-Test.

In [8]:
sales_BDM=df['Sales_before_digital_add(in $)']
sales_ADM=df['Sales_After_digital_add(in $)']

In [10]:
tvalue, pvalue=stats.ttest_rel(sales_ADM,sales_BDM,alternative='greater')
print("t value : ", round(tvalue, 4))
print("p value based on the t score is: ", round(pvalue,4))

t value :  12.0907
p value based on the t score is:  0.0


In [11]:
stats.ttest_rel(sales_ADM,sales_BDM,alternative='greater')

Ttest_relResult(statistic=12.09070525287017, pvalue=3.168333502287889e-11)

From above T test, p-value is 0 and statistical value is 12.0907.
As the p-value=0.0< α=0.05 and hence we reject the null Hypothesis 
ie. The sales after stepping into digital marketing is more.

# 2. The company needs to check whether there is any dependency between the fetures “Region” and “Manager”

# Chi Square test for Dependency

In [13]:
df3=pd.crosstab(df.Region, df.Manager, margins=True,values=df['Sales_before_digital_add(in $)'],aggfunc=np.sum)
df3

Manager     Manager - A  Manager - B  Manager - C      All
Region                                                    
Region - A       594514       466851       420684  1482049
Region - B       570900       152167       330598  1053665
Region - C       147463       469477       130625   747565
All             1312877      1088495       881907  3283279

In [14]:
valueBDM = np.array([df3.iloc[0][0:3].values,
                  df3.iloc[1][0:3].values,
                  df3.iloc[2][0:3].values])
stat, p, dof, expected=chi2_contingency(valueBDM)

In [15]:
print(chi2_contingency(valueBDM)[0:3])
alpha = 0.05
print("p value is " + str(p))
if p <= alpha:
    print('Dependent (reject H0)')
else:
    print('Independent (H0 holds true)')

(474356.9740349385, 0.0, 4)
p value is 0.0
Dependent (reject H0)


In [16]:
df4=pd.crosstab(df.Region, df.Manager, margins=True,values=df['Sales_After_digital_add(in $)'],aggfunc=np.sum)
df4

Manager     Manager - A  Manager - B  Manager - C      All
Region                                                    
Region - A      1030437       656832       701262  2388531
Region - B       939851       231808       429436  1601095
Region - C       229336       643654       222106  1095096
All             2199624      1532294      1352804  5084722

In [17]:
valueADM = np.array([df4.iloc[0][0:3].values,
                  df4.iloc[1][0:3].values,
                  df4.iloc[2][0:3].values])
stat, p, dof, expected=chi2_contingency(valueADM)

In [18]:
print(chi2_contingency(valueADM)[0:3])
alpha = 0.05
print("p value is " + str(p))
if p <= alpha:
    print('Dependent (reject H0)')
else:
    print('Independent (H0 holds true)')

(671476.3637089065, 0.0, 4)
p value is 0.0
Dependent (reject H0)


From above two cases of considering both with and wihout Digital marketing, p-value is 0 for both cases ,474356.97 and 671476.36 is the statistical value respectively and 4 is the degree of freedom.

As the p-value=0.0< 𝛼 =0.05 and hence we reject the NULL Hypothesis.

Ie. The 2 features “Region” and “Manager” which are to be compared are dependent.